File: LeetCode/1217. Minimum Cost to Move Chips to The Same Position/solution.py
"""
51 / 51 test cases passed.
Runtime: 48 ms
Memory Usage: 14.8 MB
"""
class Solution:
    def minCostToMoveChips(self, position: List[int]) -> int:
        cnt = Counter(x & 1 for x in position)
        return min(cnt[0], cnt[1])

File: LeetCode/1217. Minimum Cost to Move Chips to The Same Position/solution.cpp
/**
 * 51 / 51 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 7.1 MB 
 */
class Solution {
public:
    int minCostToMoveChips(vector<int>& position) {
        int counter[2] = {0, 0};
        for (auto &x : position) {
            ++ counter[x & 1];
        }
        return min(counter[0], counter[1]);
    }
};
